File: LeetCode/LeetCode Practice Problems/110- Balanced Binary Tree.cpp
class Solution {
public:
    bool flag = true;
    int find(TreeNode* root) {
        if(root == NULL) {
            return 0;
        }
        
        int left = find(root->left);
        int right = find(root->right);
        
        if(abs(left - right) > 1) {
            flag = false;
        }
        
        return max(left , right) + 1;
    }
    
    bool isBalanced(TreeNode* root) {
        flag = true;
        
        find(root);
        
        return flag;
    }
};

File: LeetCode/LeetCode Practice Problems/110-Balanced Binary Tree.java
class Solution {
    boolean balanced = true;
    public int height(TreeNode root) {
        if(root == null) return 0;
        int left = height(root.left);
        int right = height(root.right);
        if(Math.abs(left - right) > 1) {
            balanced = false;
        }
        return Math.max(left , right) + 1;
    }
    public boolean isBalanced(TreeNode root) {
        height(root);
        return balanced;
    }
}
